libtenzir/builtins/operators/sockets.cpp

```diff
@@ -50,12 +51,21 @@ class sockets_operator final : public crtp_operator<sockets_operator> {
   }
 };
 
-class plugin final : public virtual operator_plugin<sockets_operator> {
+class plugin final : public virtual operator_plugin<sockets_operator>,
+                     public virtual operator_factory_plugin {
 public:
   auto signature() const -> operator_signature override {
     return {.source = true};
   }
 
+  auto make(invocation inv, session ctx) const
+    -> failure_or<operator_ptr> override {
+    // XXX: Is this necessary even when there are no args? How does it even know
+    // how to call this?
```

Comment:
The argument parser here just adds a nice helptext if you try to use this with any arguments. You could also do that manually, but the argument parser has a good and somewhat standardized default.